src/main/java/teammates/storage/api/FeedbackResponsesDb.java

```diff
@@ -532,7 +534,9 @@ public FeedbackResponse getFeedbackResponseEntityOptimized (FeedbackResponseAttr
                 new ArrayList<FeedbackResponseAttributes>();
         
         for (FeedbackResponse fr : frList) {
+            if (!JDOHelper.isDeleted(fr)) {
```

Comment:
Actually, I think it might make sense to do this check for every database read, but should be discussed and done in another issue if that's the case.